lib/circuit.js

```diff
@@ -557,22 +584,27 @@ class CircuitBreaker extends EventEmitter {
     }
     const args = Array.prototype.slice.call(rest);
 
+    const cacheKey = this.options.cacheGetKey.apply(this, rest);
+
     /**
      * Emitted when the circuit breaker action is executed
      * @event CircuitBreaker#fire
      * @type {any} the arguments passed to the fired function
      */
     this.emit('fire', args);
 
-    if (CACHE.get(this) !== undefined) {
-      /**
-       * Emitted when the circuit breaker is using the cache
-       * and finds a value.
-       * @event CircuitBreaker#cacheHit
-       */
-      this.emit('cacheHit');
-      return CACHE.get(this);
-    } else if (this.options.cache) {
+    // If cache is enabled, check if we have a cached value
+    if (this.options.cache) {
```

Comment:
Done in [c2e54b8](https://github.com/nodeshift/opossum/pull/739/commits/c2e54b8f4b6b9e3fddf22d0ce46b610884151eff). Still need to have variable, because it used lower in storing value in cache.